crates/milli/src/search/facet/facet_distribution.rs

```diff
@@ -385,6 +357,61 @@ impl<'a> FacetDistribution<'a> {
 
         Ok(distribution)
     }
+
+    /// Select a field if it is faceted and in the facets.
+    fn select_field(
+        &self,
+        name: &str,
+        metadata: &Metadata,
+        filterable_attributes_rules: &[FilterableAttributesRule],
+    ) -> bool {
+        match &self.facets {
+            Some(facets) => {
+                // The list of facets provided by the user is a legacy pattern ("dog.age" must be selected with "dog").
```

Comment:
can this create an inconsistency, if the user uses the new rules syntax, and has set only `dog` as filterable? On the indexing side, `dog.age` might not be filterable, but here we will return it as filterable.